src/XML/src/FMI3/fmi3_xml_model_structure.c

```diff
@@ -442,67 +406,69 @@ int fmi3_xml_parse_unknown(fmi3_xml_parser_context_t *context,
         return -1;
     }
 
-    return fmi3_xml_parse_dependencies(context, parentElmID, deps);
+    return fmi3_xml_parse_dependencies(context, elmID, deps);
 }
 
 
-/*int fmi3_xml_handle_OutputUnknown(fmi3_xml_parser_context_t *context, const char* data) {*/
+// TODO: Currently required for structural purposes, but not a valid ModelStructure element on its own
```

Comment:
This is something that might be desirable to fix here, not sure what would be the best way.

Reply:
Add an assert(0) as well and we're good.

For some other elements that can appear on multiple places, the one that happens first in the XML (e.g. Real under TypeDefinitions in FMI2), is the handler for that element (..._handle_Real), while the ones that happen later need to name something like (..._handle_RealVariable). I think that's more confusing.

I don't think it's worth the effort trying to rewrite it

Reply:
Resolved this by instead removing the fmi3_xml_scheme_Unknown instead. It was essentially an abstract type and that behavior is somewhat kept by having an internal shared parsing function.

Note: Unknown elements in ModelStructure are now skipped (with log message).